src/components/Dashboard/RewardsModule.js

```diff
@@ -31,50 +33,72 @@ const useTotalDisputesFees = (arbitrableFees, appealFees) => {
 // Dispute fees + Appeal Fees => DAI fees
 const RewardsModule = React.memo(function RewardsModule({
   rewards,
+  treasury,
   loading,
+  onWithdraw,
   onSettleReward,
   onSettleAppealDeposit,
 }) {
   const wallet = useWallet()
+  const { feeToken } = useCourtConfig()
+
+  const { totalArbitrableFees, totalAppealFees } = useTotalDisputesFees(
+    rewards?.arbitrableFees,
+    rewards?.appealFees
+  )
+
+  // TODO: Handle possible multiple tokens (fee token can change)
+  const treasuryToken = treasury?.find(({ token }) =>
+    addressesEqual(token.id, feeToken.id)
+  )
+  const treasuryBalance = treasuryToken ? treasuryToken.balance : bigNum(0)
+  const totalDisputesFees = totalArbitrableFees.add(totalAppealFees)
+  const totalFees = totalDisputesFees.add(treasuryBalance)
 
   // Form submission
   const handleFormSubmit = async event => {
     event.preventDefault()
 
-    if (!rewards) return
+    const claimRewards = async () => {
+      if (!rewards) return
 
-    try {
-      // Claim all arbitrable fee rewards
-      const arbitrableTxs = []
-      for (const arbitrableFee of rewards.arbitrableFees) {
-        const { disputeId, rounds } = arbitrableFee
-        for (const roundId of rounds) {
-          arbitrableTxs.push(
-            await onSettleReward(disputeId, roundId, wallet.account)
-          )
+      const transactionBag = []
```

Comment:
Don't know if we really want to call this a "bag" (it's an awkward bit of history from the client).

Perhaps `rewardTransactionQueue`?